### PR TITLE
simplify resolve()

### PR DESCRIPTION
**Pull request by WalterBright**
A nice shrink on the # of lines to implement it.

### CODE REVIEW

**Review by thewilsonator — APPROVED** [review]

File: src/dmd/typesem.d [hunk 1]
```diff
@@ -2783,56 +2775,49 @@ void resolve(Type mt, const ref Loc loc, Scope* sc, Expression* pe, Type* pt, Ds
                 e = e.expressionSemantic(sc);
                 mt.resolveExp(e, pt, pe, ps);
             }
+            if (*pt)
+                (*pt) = (*pt).addMod(mt.mod);
         }
-        if (*pt)
-            (*pt) = (*pt).addMod(mt.mod);
```
Comment (ibuclaw) [inline src/dmd/typesem.d]: Why was this moved?

Comment (WalterBright) [reply src/dmd/typesem.d]: It was done differently on line 2766 of the PR.

**Review by ibuclaw — APPROVED** [review]
